src/algo/dominators.rs

```diff
@@ -76,6 +79,52 @@ impl<N> Dominators<N>
             None
         }
     }
+
+    /// Returns the dominance frontier of every node in `G`.
+    ///
+    /// From Cytron et. al, the dominance frontier is defined as:
+    /// > the set of all cfg nodes, y, such that b dominates a predecessor of y but does not strictly
+    /// > dominate y
+    ///
+    /// We use the algorithm detailed here: https://www.cs.rice.edu/~keith/EMBED/dom.pdf
+    pub fn dominance_frontiers<G>(&self, graph: G) -> HashMap<G::NodeId, Vec<G::NodeId>>
```

Comment:
I think we should return a newtype over the hash map, `DominanceFrontiers`, and expose semantically meaningful operations on that newtype.